A79-non-per-call-metrics-architecture.md

```diff
@@ -0,0 +1,598 @@
+```java
+/**
+ * Registry for metric instruments (also known as metric descriptors).
+ */
+@Internal
+public class MetricInstrumentRegistry {
+  
+  // Returns a default MetricInstrumentRegistry instance
+  public static MetricInstrumentRegistry getInstance();
+  
+  // Returns a list of registered metric descriptors.
+  public List<MetricDescriptor> getMetricInstruments();
+  
+  // Register and return a long counter metric descriptor for recording long
+  // counter values.
+  public LongCounterDescriptor registerLongCounter(
+      String name, String description, String unit, List<String> labelKeys, 
+      List<String> optionalLabelKeys, boolean isEnabledByDefault);
+
+  // Register and return a double histogram metric descriptor for recording 
+  // double histogram values.
+  public DoubleHistogramDescriptor registerDoubleHistogram(
+      String name, String description, String unit, List<Double> bucketBoundaries,
+      List<String> labelKeys, List<String> optionalLabelKeys, 
+      boolean isEnabledByDefault);
+}
+
+interface MetricDescriptor {
+  long index;
+  default long getIndex() { return index; }
+}
+
+class LongCounterDescriptor implements MetricDescriptor {
+  String name;
+  String description;
+  String unit;
+  List<String> labelKeys;
+  List<String> optionalLabelKeys;
+  boolean isEnabledByDefault;
+}
+
+class DoubleHistogramDescriptor implements MetricDescriptor {
+  String name;
+  String description;
+  String unit;
+  List<Double> bucketBoundaries;
+  List<String> labelKeys;
+  List<String> optionalLabelKeys;
+  boolean isEnabledByDefault;
+}
+```
+
+##### Go
+
+(To be filled.)
+
+#### Global Stats Plugin Registry
+
+A global stats plugin registry is added. This registry keeps track of all stats
+plugins that have been registered globally.
+
+The application will use a stats plugin builder (e.g.
+`OpenTelemetryPluginBuilder`) to register a stats plugin to the registry. On
+creation, stats plugins should query the global instruments registry to get the
+list of instrument descriptors. Based on these descriptors and the stats
+plugin's configuration, backing instruments (e.g., `OpenTelemetry Instruments`)
+will be created. For java, backing instruments will be lazily created. 
+Implementations can choose an alternative form of representation for the global 
+stats plugin registry. For example, gRPC Go might choose to register global 
+stats plugin through global dial options.
+
+![](A79_graphics/global-stats-plugin-registry-usage.png)
+
+##### Scoping Stats Plugins for Channels
+
+To allow stats plugins to scope themselves to a certain set of channels,
+implementations can choose one of two approaches.
+
+In the first approach, stats plugins provide a scope filter that applications
+can configure. Based on the channel's configuration, a stats plugin can decide
+whether it is interested in the channel. At channel creation time, the channel's
+target and the authority is provided to the stats plugin to make this decision.
```

Comment:
Yeah, it's the canonical target.

For the authority, it's the default authority for the channel. If the user overrides the default one for the channel, that will be used, otherwise it'll be derived from the target.